### ISSUE TITLE
[Task 2] 아파트 매매가격 예측 (4 conditions × 5 seeds)

### ISSUE THREAD

**Issue by yeongseon** [Launa-0/kpubdata-paper#14]
## Dataset

`seoul-apartment-trades`

## 예측 대상

`price_per_m2`

## Feature 예

```
district, area_m2, year, month, floor, build_year,
historical transaction features
```

## 모델

모델은 연구의 핵심이 아니므로 **단순하고 재현 가능한 모델 1종**을 사용한다.
후보: Linear Regression / Random Forest / LightGBM

**동일 모델, 동일 hyperparameter를 모든 조건에 적용한다.**

## 실험 프로토콜

```
random seeds: 5
보고: mean ± standard deviation, 95% confidence interval
```

## 측정

- RMSE
- MAE
- training failure rate
- preprocessing LOC / steps / runtime

## 완료 조건

- [ ] 모델 및 hyperparameter 확정 (선택 근거 기록)
- [ ] train/test 분할 전략 확정 (시간 기반 분할 권장 — leakage 방지)
- [ ] 4개 조건 × 5 seeds 실행
- [ ] 결과 기록 및 Table 4 반영

**Comment by Launa-0** [comment]
**시드를 5개에서 6개로 올리는 것을 검토해 주세요.** #21을 구현하면서 나온 결과입니다.

**n=5에서는 양측 Wilcoxon signed-rank 검정이 p<0.05에 도달할 수 없습니다.** 부호 배정이 2⁵가지뿐이라 최소 달성 가능한 p가 2/2⁵ = 0.0625입니다. 효과 크기와 무관합니다.

| n | 최소 달성 가능한 양측 p | 0.05 도달 |
| ---: | ---: | :--- |
| 4 | 0.1250 | 불가 |
| **5** | **0.0625** | **불가** |
| 6 | 0.0312 | 가능 |

paired t-검정에는 이런 하한이 없습니다 — 같은 5쌍이 t로는 p≈0.013에 도달합니다. 그래서 **이 표본 크기에서는 유의성 도달 가능 여부를 효과 크기가 아니라 정규성 검정이 결정합니다.** 차이가 치우쳐 있으면 Wilcoxon으로 가서 상한에 걸리고, 얌전하면 t로 가서 안 걸립니다.

측정하려는 파이프라인의 성질이 아니라 **실험 설계의 성질**입니다. 조건당 시드 하나를 더 돌리는 비용이 이 제약을 없애는 가장 싼 방법입니다.

#21에서 할 수 있는 건 다 했습니다 — `significance_reachable`을 모든 비교와 요약표에 실어서 `p=0.0625`가 귀무 결과로 오독되지 않게 했고, 효과 크기·신뢰구간·원자료를 함께 보고합니다. 다만 이건 **완화이지 해결이 아닙니다.** 결정은 이 이슈(과제 설계)에 속합니다.

5개를 유지한다면 그 선택과 근거를 Threats to Validity에 적어야 합니다 — 초안은 [`docs/statistics.md`](../blob/main/experiments/docs/statistics.md)에 있습니다.

**Comment by yeongseon** [comment]
이 이슈 진행합니다 (Task 2, 가격 예측). Task 1/3과 같은 구조를 따릅니다.

**Comment by yeongseon** [comment]
이 이슈 착수했다가 중단합니다. 다른 분이 가져가셔도 됩니다.

시작하자마자 걸린 구조적 문제 하나를 적어 둡니다 — **현재 seed가 분석 함수까지 닿지 않습니다.**

```python
# runner.py
prepared = runner.prepare(context)
return prepared, task.analyze(prepared)   # ← seed가 안 넘어감
```

`RunContext.seed`는 `prepare()` 에만 전달되고, `Task.analyze` 시그니처는 `Callable[[AnalysisInput], AnalysisOutput]` 입니다. 그런데 이 과제는 **5 seed × 4 조건**이고 모델 적합은 분석 쪽에 있어야 합니다 — `prepare()` 안에서 모델을 적합시키면 모델 비용이 RQ2(전처리 비용)로 청구되고, 조건마다 다른 모델을 적합시킬 수 있게 됩니다.

제안하던 수정은 시그니처를 `analyze(prepared, *, seed: int = 0)` 으로 통일하고 runner가 `task.analyze(prepared, seed=seed)` 로 부르는 것입니다. task01/task03은 인자를 받고 무시합니다(결정적 분석). introspection 없이 명시적이고, 기존 테스트의 `analysis.analyze(prepared)` 호출은 기본값으로 그대로 동작합니다.

함께 정해야 할 것:

- **분할은 시간 기준이어야 합니다.** 무작위 분할 + 과거 거래 feature는 leakage입니다. train ≤2023 / test 2024 같은 형태.
- **과거 거래 feature를 넣으면 Gold가 실제로 유리해집니다** — Task 1의 Gold(자치구×월 집계)가 그대로 feature가 됩니다. 예측 시점보다 엄격히 앞선 달만 쓰면 leakage가 없습니다.
- `n_jobs=1` 로 두어야 재현됩니다.

작업물은 커밋하지 않았습니다 (소비자 없는 시그니처 변경만 남아서).

**Comment by seoL-ee** [comment]
**원래 목적** 아파트 매매가격 예측을 4조건 × 5 seeds로 수행해 계층별 준비 비용과 예측 오차를 비교하는 것이었다.

**변경 이유** 모델 선택·temporal split·seed variance가 Medallion 효과가 아닌 ML 실험 설계 면적을 늘리고, 같은 데이터셋을 T1이 이미 쓰고 있어 파이프라인 construct를 추가로 검증하지 못한다.

**최종 처리** scope에서 제외한다. downstream task는 T1·T3만 남는다.

상세: [experiment-design-revisions.md](https://github.com/Launa-0/kpubdata-paper/blob/de25c39/experiments/docs/experiment-design-revisions.md) R-7.
